k8s/elasticsearch/apptest/tester/tests/basic-suite.yaml

```diff
@@ -1,8 +1,89 @@
 actions:
-- name: dummy test
+- name: Wait until all expected nodes are connected to the cluster
   bashTest:
     script: |-
-      exit 0
+      function get_nodes_count() {
+        curl -s "${HEALTH_URL}" | jq '.number_of_nodes'
+      }
+
+      expected_nodes="${REPLICAS}"
+
+      timeout=300
+      timeout_time=$(( $(date +%s) + timeout ))
+
+      # Wait for all nodes to join the cluster
+      nodes_count="$(get_nodes_count)"
+      while [[ "${nodes_count}" -ne "${expected_nodes}" ]] \
+          && [[ "$(date +%s)" -le "${timeout_time}" ]]; do
+        sleep 3
+        nodes_count="$(get_nodes_count)"
+      done
+
+      if [[ "${nodes_count}" -eq "${expected_nodes}" ]]; then
+        echo "OK - all nodes joined"
```

Comment:
Can we simply have if statement saying if not equal then `exit 1`, and modify the `stdout` to expect `exitCode` instead?

Reply:
I prefer stdout checks for ease of debugging in case of failure, even if I agree that checking the exitCode is more readable. We can return to checking the exitCode once testrunner prints the logs from both stdout and stderr in case of failure.